src/main/cpp/util/md5.h

```diff
@@ -24,6 +24,15 @@
 #define __alignof__ __alignof
 #endif  // _MSC_VER && !__alignof__
 
+#if defined(__linux__)
+#include <endian.h>
+#elif defined(__FreeBSD__) || defined(__OpenBSD__)
+#include <sys/endian.h>
+#elif defined(__APPLE__) || defined(_WIN32)
+// Hopefully OSX and Windows will keep running solely on little endian CPUs, so:
```

Comment:
Wouldn't this cause VERY subtle bugs, should this assumption ever change? Not that anyone likes big-endian anymore. Do these macros not exist on Mac OS and Windows?

Maybe put in an endianness check in the constructor or a static initializer if you can't find the equivalent logic on Windows and Mac OS?

Reply:
It will cause great big bugs if that is not true. And the md5 test will catch those.
So I am not too worried. There is an apple mitigation, in that they have an endian.h file somewhere in a framework. I did not want to modify our build too much now for something we may never need.

Windows has declared they are little endian. My limited research said that choice was from 1989.